src/Microsoft.AspNetCore.Server.Kestrel/Http/Connection.cs

```diff
@@ -32,23 +32,20 @@ public class Connection : ConnectionContext, IConnectionControl
         private readonly object _stateLock = new object();
         private ConnectionState _connectionState;
 
-        private IPEndPoint _remoteEndPoint;
-        private IPEndPoint _localEndPoint;
-
         public Connection(ListenerContext context, UvStreamHandle socket) : base(context)
         {
             _socket = socket;
             ConnectionControl = this;
 
-            _connectionId = Interlocked.Increment(ref _lastConnectionId);
+            ConnectionId = Interlocked.Increment(ref _lastConnectionId).ToString(CultureInfo.InvariantCulture);
```

Comment:
As this is a property could it be lazy-init? Though might not work with the logging; as will currently always init.

Logging could be passed a `ConnectionContext` rather than `ConnectionId` and if enabled at that level then call `ConnectionId` which inits the value.

`Task.Id` in coreclr uses this approach https://github.com/dotnet/coreclr/pull/3139